frontend-react/src/utils/Analytics.ts

```diff
@@ -0,0 +1,19 @@
+import { IEventTelemetry } from "@microsoft/applicationinsights-web";
+
+import { getAppInsights } from "../TelemetryService";
+
+export enum EventName {
+    TABLE_FILTER = "Table Filter",
+}
+
+export const trackAppInsightEvent = (
+    eventName: string = "",
+    eventData: any
+) => {
+    const appInsights = getAppInsights();
+    const telemetryEvent: IEventTelemetry = {
+        name: eventName,
+    };
+
+    if (eventName !== "") appInsights?.trackEvent(telemetryEvent, eventData);
```

Comment:
If we make `eventName` a required string/remove the fallback [above](https://github.com/CDCgov/prime-reportstream/pull/7805/files#diff-6fd707ed4d605aa3f4eec2edc8899327ec74ed71587dffa749987e5a7ea591ddR10), we shouldn't need to check for an empty string.